### PR TITLE
[Doc] Update Query Rewrite Desc for 2.5.1

### PR DESCRIPTION
**Pull request by EsoragotoSpirit**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Review by amber-create — COMMENTED** [review]
image
改为 -> StarRocks supports rewriting queries on materialized views that are created on the default catalog or an external catalog such as a Hive catalog, Hudi catalog, or Iceberg catalog.

**Review by amber-create — COMMENTED** [review]
image
-> data changes

File: docs/using_starrocks/Materialized_view.md [hunk 1]
```diff
@@ -201,15 +201,11 @@ MySQL > SELECT * FROM order_mv;
...
-
-> **CAUTION**
->
-> Currently, StarRocks does not support query rewrite based on external catalog materialized views.
+Currently, StarRocks supports query rewrite based on materialized views on the default catalog, the Hive catalog, the Hudi catalog, and the Iceberg catalog. When querying data in the default catalog, StarRocks ensures strong consistency of results between the rewritten query and the original query by excluding materialized views whose data is inconsistent with the base table. When the data in a materialized view data expires, the materialized view will not be used as a [candidate materialized view](#candidate-materialized-view-for-query-rewrite). When querying data in external catalogs, StarRocks does not ensure strong consistency of the results because StarRocks cannot perceive the data change in partitions of external catalogs.
```
Comment (evelynzhaojie) [inline docs/using_starrocks/Materialized_view.md:204]: When the data in a materialized view data --- 最后一个data多余

**Review by evelynzhaojie — APPROVED** [review]

**Review by amber-create — APPROVED** [review]

**Comment by EsoragotoSpirit** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**